apps/meteor/ee/app/models/server/raw/LivechatRooms.ts

```diff
@@ -0,0 +1,78 @@
+import { LivechatRoomsRaw } from '../../../../../server/models/raw/LivechatRooms';
+import { queriesLogger } from '../../../livechat-enterprise/server/lib/logger';
+import { addQueryRestrictionsToRoomsModel } from '../../../livechat-enterprise/server/lib/query.helper';
+import { overwriteClassOnLicense } from '../../../license/server';
+
+const applyRestrictions = (method: 'find' | 'update' | 'findPaginated') =>
+	function (originalFn: any, originalQuery: object, ...args: any) {
+		const query = addQueryRestrictionsToRoomsModel(originalQuery);
+		queriesLogger.debug({ msg: `LivechatRoomsRaw.${method}`, query });
+
+		// @ts-expect-error had to manually override the type because of the way the args are passed
+		return originalFn.call(this, query, ...args);
+	};
+
+overwriteClassOnLicense('livechat-enterprise', LivechatRoomsRaw, {
+	find: applyRestrictions('find'),
+	update: applyRestrictions('update'),
+	findPaginated: applyRestrictions('findPaginated'),
+	updateDepartmentAncestorsById(_, _id, departmentAncestors) {
+		const query = {
+			_id,
+		};
+		const update = departmentAncestors ? { $set: { departmentAncestors } } : { $unset: { departmentAncestors: 1 } };
+		return this.update(query, update);
+	},
+	async associateRoomsWithDepartmentToUnit(_, departments: string[], unitId: string) {
+		const query = {
+			$and: [
+				{
+					departmentId: { $in: departments },
+				},
+				{
+					$or: [
+						{
+							departmentAncestors: { $exists: false },
+						},
+						{
+							$and: [
+								{
+									departmentAncestors: { $exists: true },
+								},
+								{
+									departmentAncestors: { $ne: unitId },
+								},
+							],
+						},
+					],
+				},
+			],
+		};
+		const update = { $set: { departmentAncestors: [unitId] } };
+		queriesLogger.error({ msg: `LivechatRoomsRaw.associateRoomsWithDepartmentToUnit - association step`, query, update });
```

Comment:
Do we need these logs?

Reply:
Oops! They were supposed to be just debug logs :grimacing: Changed them back to debug :heavy_check_mark: